client/src/TenderList.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect } from "react";
+
+const TendersList = () => {
+	const [tenders, setTenders] = useState([]);
+	const [loading, setLoading] = useState(true);
+	const [page, setPage] = useState(1);
+	const [hasMore, setHasMore] = useState(true);
+
+	const fetchTenders = async (pageNumber) => {
+		setLoading(true);
+		try {
+			const response = await fetch(`/api/tenders?page=${pageNumber}`);
+			const data = await response.json();
+			setTenders((prevTenders) => [...prevTenders, ...data.tenders]);
+			setHasMore(data.tenders.length === 25);
+		} catch (error) {
+			console.error("Error fetching tenders:", error);
+		} finally {
+			setLoading(false);
+		}
+	};
+
+	useEffect(() => {
+		fetchTenders(page);
+	}, [page]);
+
+	const loadMore = () => {
+		if (hasMore && !loading) {
+			setPage((prevPage) => prevPage + 1);
+		}
+	};
+
+	return (
+		<div>
+			<h2>Tenders List</h2>
+			<table>
+				<thead>
+					<tr>
+						<th>Tender ID</th>
+						<th>Tender Title</th>
+						<th>Tender Created Date</th>
+						<th>Tender Announcement Date</th>
+						<th>Tender Closing Date</th>
+						<th>Tender Project Deadline Date</th>
+						<th>Tender Status</th>
+					</tr>
+				</thead>
+				<tbody>
+					{tenders.map((tender) => (
+						<tr key={tender["Tender ID"]}>
+							<td>{tender["Tender ID"]}</td>
+							<td>{tender["Tender Title"]}</td>
+							<td>
+								{new Date(tender["Tender Created Date"]).toLocaleDateString()}
+							</td>
+							<td>
+								{new Date(
+									tender["Tender Announcement Date"]
+								).toLocaleDateString()}
+							</td>
+							<td>
+								{new Date(tender["Tender Closing Date"]).toLocaleDateString()}
+							</td>
+							<td>
+								{new Date(
+									tender["Tender Project Deadline Date"]
+								).toLocaleDateString()}
+							</td>
+							<td>{tender["Tender Status"]}</td>
+						</tr>
+					))}
+				</tbody>
+			</table>
+			{loading && <p>Loading...</p>}
+			{hasMore && !loading && <button onClick={loadMore}>Load More</button>}
```

Comment:
Load more, makes it sound like more items are going to be added inline. I think `next page` is clearer

Reply:
I have changed it to Next Page